### PR TITLE
Added Documentation on Guidelines to Contribute, CODE_OF_CONDUCT.md and SECURITY.md

### PR DESCRIPTION
**Pull request by Keerthivasan-Venkitajalam**
This PR addresses issue #3 by adding documentation on contributing guidelines as outlined. The following changes have been made:  

1. **Edited `README.md`:**
   - Added a new section to outline the contribution guidelines.
   - Detailed steps for contributors to set up the repository locally, create branches, make changes, and submit PRs.

2. **Created `CODE_OF_CONDUCT.md`:**  
   - Outlines the expected behavior of contributors and community members.  
   - Includes guidance on reporting unacceptable behavior.  

3. **Created `SECURITY.md`:**  
   - Provides instructions for reporting security vulnerabilities.  
   - Ensures contributors and users know how to handle security concerns responsibly.  

4. **Ensured All Files Are in the `/docs` Folder:**  
   - Placed `CODE_OF_CONDUCT.md`, and `SECURITY.md` in the `/docs` directory as per the issue requirements.  

---

### **Additional Notes:**  
- All documentation is consistent with GitHub community standards.  
- The content is written with clarity and accessibility in mind, ensuring ease of use for new and experienced contributors alike.  

Please review the changes and let me know if further modifications are required. 😊

### CODE REVIEW

**Review by Ashrockzzz2003 — CHANGES_REQUESTED** [review]
Please move the details on contributing to `docs/CONTRIBUTING.md`. 
Will review the rest after it's done. Please also take a look at https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors

**Review by VishalTheHuman — COMMENTED** [review]
Make the changes mentioned and create another pull request @Keerthivasan-Venkitajalam

**Comment by Keerthivasan-Venkitajalam** [comment]
> Make the changes mentioned and create another pull request @Keerthivasan-Venkitajalam

Got it will do that... Thank you.

**Comment by VishalTheHuman** [comment]
@Keerthivasan-Venkitajalam No need to create a new PR. Keep this one itself.

**Comment by Keerthivasan-Venkitajalam** [comment]
> @Keerthivasan-Venkitajalam No need to create a new PR. Keep this one itself.

Okay Anna... got it.

**Comment by Keerthivasan-Venkitajalam** [comment]
> Make the changes mentioned and create another pull request @Keerthivasan-Venkitajalam

I have completed the requested changes @VishalTheHuman @Ashrockzzz2003

File: README.md [hunk 1]
```diff
@@ -1,20 +1,151 @@
...
+    <img align='center' src="https://img.shields.io/badge/LICENSE-GPL-green"/>
+  </a>
+  <a href="https://github.com/Ashrockzzz2003/Data_Structures_and_Algorithms" target="_blank" rel="noreferrer">
+    <img align='center' src="https://img.shields.io/github/created-at/Ashrockzzz2003/Data_Structures_and_Algorithms"/>
+  </a>  
+</div>
 
-<!--
-## Contributions required on
+Welcome to the **Data Structures and Algorithms** repository—a comprehensive collection of foundational data structures and algorithms implemented in **Python**, **C++**, **Go**, and **Java**. This repository serves as a resource hub for **computer science students** and **enthusiasts** eager to deepen their understanding of algorithmic design.
```
Comment (Ashrockzzz2003) [inline README.md:12]: I want the old description to be there.
Don't want this one. Feels very AI generated.

File: README.md [hunk 2]
```diff
@@ -1,20 +1,151 @@
...
+
+---
+
+### **Why Use This Repository?**
+
+- 📘 **Educational Resource**: Ideal for mastering fundamental concepts in computer science.
+- 🌍 **Community Collaboration**: Contributions from students and developers create a dynamic and growing repository.
+
+---
+
```
Comment (Ashrockzzz2003) [inline README.md:48]: overview and why this repo not needed.
Remove contributing data from README.

Instead, just have a link to CONTRIBUTING.md.

File: docs/SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Security Policy
+
+## Supported Versions
+
+| Version | Supported           |
+|---------|---------------------|
+| v1.0.x  | ✅ Full support     |
```
Comment (Ashrockzzz2003) [inline docs/SECURITY.md:7]: We have only one version.
v1.0.x as of now.

**Review by Ashrockzzz2003 — CHANGES_REQUESTED** [review]
I feel the content to be very AI generated. Can be made better with spending some time to this. Take your time and then continue working on the PR.

**Review by Ashrockzzz2003 — CHANGES_REQUESTED** [review]
image

Feels Repitive. The title `Data Structures and Algorithms`

**Comment by Keerthivasan-Venkitajalam** [comment]
> image
> 
> Feels Repitive. The title `Data Structures and Algorithms`

Changes made.

File: README.md [hunk 4]
```diff
@@ -1,20 +1,116 @@
...
+## **Contributors**
+
+| **Contribution Type**    | **Contributor**      |
+| ------------------------ | -------------------- |
+| Python Implementation   | Ashwin Narayanan S   |
+| C++ Implementation      | Aasil                |
+
```
Comment (Ashrockzzz2003) [inline README.md:99]: The word `implementation` sounds odd.
Also, probably link `Contributor` names to GitHub account URL.

**Review by Ashrockzzz2003 — CHANGES_REQUESTED** [review]
Instead of having most of the content in **bold**, we can have it in `tags` like this to highlight. Just a suggestion.

File: README.md (hunk 4 above)
Comment (Keerthivasan-Venkitajalam) [reply README.md:99]: Will do that anna @Ashrockzzz2003

**Comment by Keerthivasan-Venkitajalam** [comment]
> Instead of having most of the content in **bold**, we can have it in `tags` like this to highlight. Just a suggestion.

Changes made and the contribution table has been modified according to the feedback.

**Review by Ashrockzzz2003 — APPROVED** [review]
LGTM

**Comment by Ashrockzzz2003** [comment]
Thanks @Keerthivasan-Venkitajalam for the contribution.

**Pull request merged**